### DISCUSSION TITLE
Dips in ET and Stack Temp

### DISCUSSION THREAD

**Discussion by alikaMCR** [artisan-roaster-scope/artisan#874] [Q&A] [answered]
Hello everyone,
First of all, I'm not very competent at this type of troubleshooting or even these types of forum discussion.  This is all very much over my head, so bear with me as I try to understand these concepts.
I am currently using a Loring s35.  We recently purchased a new work laptop for monitoring our roast profiles.  I downloaded the latest version of artisan and noticed that I will have these little dips in communication loss in the ET which also affect my ETROR.  After a little bit of digging around in the Roast Properties - Data, I noticed that my stack temperature will display as -1 or u.u on the main display.  However, sometimes it will display a reading, and when it does, I lose another temperature reading.  Typically, this is the ET.  Then at some point, usually quickly after, it reverts back to normal.  On the previous laptop, we never had an issue like this.  I attached a couple images of what I noticed.  
Can anyone tell me what they think is going on?  Any solutions?
Thank you very much.
<img width="959" alt="Big dips" src="https://user-images.githubusercontent.com/104863062/166567123-e6f4732a-5db7-4fd8-a948-f476f3cac4ee.png">
<img width="490" alt="Big dips data" src="https://user-images.githubusercontent.com/104863062/166567151-d8abefe5-855b-4b2a-a171-5f7eeecbde1c.png">

-Alika

**Comment by poundy** [comment]
u.u or -1.0 means the measurement is unavailable. This usually means some form of communication issue between the roaster and the laptop.  

Not knowing anything about the Loring you're roasting on, and knowing the laptop is the only thing that has changed, points to the laptop or it's OS or drivers, that might be coming in to play here. You haven't mentioned these details, can you elaborate on the working vs not-working laptop and these factors? 

One common issue that is seen with laptops and (usually) thermocouples is ground reference plane differences. Often that situation can be improved by disconnecting the laptop power supply so that it runs from battery, and sometimes it is necessary to replace the USB cable with a properly electrically isolated one. Can you test doing a roast from battery and see if that helps?

**Reply by poundy** [reply]
also, can you show us the ET data for that period that it drops on the graph? I'd like to see the point just before the stack temp goes AWOL.

**Comment by alikaMCR** [comment]
First, thanks for such a quick response.  
I don't have an awful lot of detail between the differences in the previous laptop and our current one.  The only difference I know for sure is that the previous laptop ran Windows 10 where as our new one is running Windows 11.  Prompted by your question I made sure to check all my drivers are updated.  Which they seem to be. 
To elaborate on your power supply comment.  When researching this issue previously, I had noticed a post about ground interference, so I haven't plugged in my laptop while roasting.  This didnt appear to fix the issue.  
I have also attached a better snip of the Roast Properties data that shows the moments before the ET got wonky.  Hopefully you can extrapolate from that.
Appreciate your help.  Thank you
<img width="702" alt="ET pre dip" src="https://user-images.githubusercontent.com/104863062/166588386-f64d5528-6a1f-4cfd-8db8-0a1228b455f2.png">

**Reply by poundy** [reply]
all good - the community here are here to help where possible ! 

All the data above shows "stack" as -1.0 and no valid data, so i don't think it's showing anything of value for that problem.  You however say the ET went wonky, but in my mind there's nothing wrong with that, it isn't "normal" but it also didn't show any obvious issue with it or the other data (besides being unexpectedly high).  Where I really think there will be some insight is in the data table between 5 minutes and 6 minutes, as the glitch there seems to imply some other behaviour - it's almost like the roaster returns the values in the incorrect order (from a modbus perspective, if that's what the Loring uses, it may have missed a register and it's dropped the expected value in the wrong variable).

**Reply by alikaMCR** [reply]
My apologies, I totally sent the wrong clip in time.  Here is the correct data log from that period of time.  Also, for more clarity, yes I do get a "modbus communication error" occassionally.  So you seem to be right on the money. 
<img width="704" alt="miscommunication data" src="https://user-images.githubusercontent.com/104863062/166742331-8f6735f9-db50-4bc2-8bc8-561efc460654.png">

**Comment by MAKOMO** [comment]
What could help is to attach this profile as Artisan .alog file here as well as your Artisan settings on that machine as .aset file (export via menu Help >> Save Settings). Both files have to renamed to hold extension .txt, otherwise Github does not allow them to be drag-and-dropped here.

I guess we see here some unstable communication between the machine and Artisan. In the past similar issues have been seen and were caused by the PLC in the machine "going wild" if it receives too many requests. This is why I want to check your settings as we modified the communication such that Artisan would send less requests.

**Reply by alikaMCR** [reply]
Here are my settings and the particular "French" profile I've been referencing. I hope this is what you were asking for.
[French Maui Blend_22-05-03_1056.txt](https://github.com/artisan-roaster-scope/artisan/files/8628902/French.Maui.Blend_22-05-03_1056.txt)
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/8628906/artisan-settings.txt)
.

**Reply by MAKOMO** [reply]
Thanks for those settings and the profile.

Your Stack temperature is only reported after DROP as you configured an input filter (Config >> Curves, 2nd tab Filters, "Input Filter section, setting "Limits") to cut values below 50 and above 1000. For the same reason also burner readings below 50% are not registered. I suggest to untick this limit.

You did configure a quantifier to map the burner % to burner events. This does not make much sense to me and might be part of the issue. Processing those event quantifiers, and especially drawing all those events, might disturb the timing here. Why not just deactivate this one and if you want to see the burner percentage higher in the graph just lift it with a symbolic formula (x + 100 put under y1(x) for the first extra device entry).

I can only speculate about the cause of those spikes. It seems as if Artisan was requesting those 10 register values as for every sampling, but got back just the readings for 9 registers from the machine. As one was dropped, the data was incorrectly assigned to the Artisan channels. This might have been caused by a previous request from Artisan not being answered by the machine. In this case Artisan immediately sends a second request for data. That was maybe too much for the machine and it got confused. Those Loring PLCs are know for being notoriously slow in communication. Very strange for a machine that has been developed in Silicon Valley.

Alternatively you could try to increase the sampling rate to 3sec which could reduce the communication stress. I assume you directly connected Artisan to the machine via Ethernet cable and not via Wifi which can be easily disturbed by the motors of such machines.

What could help us to investigate further is your Artisan log. You could send this to info@artisan.plus by holding the ALT key while clicking on the plus icon in the top left corner of the Artisan window. An email should open with those log files attached.

**Reply by alikaMCR** [reply]
After adjusting the settings you suggested, I had a roast with no disturbance at all.  Which was really rare. Usually every roast has some kind of miscommunication.  
However, the next roasts, it continued.  Little swaps in my ET and BT.  A little frustrated, I reset my artisan settings to its factory settings.  Hoping to just start fresh.  Still no luck.    

For clarity too, I have been using wifi to connect to the Loring.  As far as I know we have never connected over ethernet.

Finally, I followed your instruction and sent my Artisan logs to that email.  Hoping there is some clue in there as to what needs to be adjusted or what's going wrong.  

Thank you for your continued efforts.  Your help is so appreciated.

**Comment by MAKOMO** [comment, marked as answer]
This has been turned into Issue #883  and resolved in the current trunk by verifying the received reply length and reject incomplete responses and re-request the data.
